File: Code 3 Callouts/Code 3 Callouts/Models/Peds/IPedBase.vb
Imports Rage

Namespace Models.Peds

    Public Interface IPedBase

        Property Type As PedType
        Property Name As String
        Property DisplayName As String
        Property Blip As Blip
        'Property SearchArea As Blip
        Property OriginalSpawnPoint As Vector3
        Property SpeechLines As List(Of String)
        Property SpeechIndex As Integer
        Property PhysicalCondition As String

        Sub Speak()
        Sub CreateBlip(Optional ByVal pColor As Drawing.Color = Nothing)
        Sub DeleteBlip()
        'Sub CreateSearchArea()
        'Sub DeleteSearchArea()

        Sub SetIsDrunk(ByVal pValue As Boolean)
        Sub AttackPed(ByVal pTargetPed As Ped)
        Sub TurnToFaceEntity(ByVal pTarget As Entity, Optional ByVal pTimeout As Integer = 5000)

    End Interface

End Namespace

File: Code 3 Callouts/Code 3 Callouts/Models/Peds/IPedBase.cs
using Rage;
using System.Collections.Generic;
using static Stealth.Plugins.Code3Callouts.Common;

namespace Stealth.Plugins.Code3Callouts.Models.Peds
{

	internal interface IPedBase
	{

		PedType Type { get; set; }
		string Name { get; set; }
		string DisplayName { get; set; }
		Blip Blip { get; set; }
		//Property SearchArea As Blip
		Vector3 OriginalSpawnPoint { get; set; }
		List<string> SpeechLines { get; set; }
		int SpeechIndex { get; set; }

		string PhysicalCondition { get; set; }
		void Speak();
		void CreateBlip(System.Drawing.Color? pColor = null);
		void DeleteBlip();
		//Sub CreateSearchArea()
		//Sub DeleteSearchArea()

		void SetIsDrunk(bool pValue);
		void AttackPed(Ped pTargetPed);

		void TurnToFaceEntity(Entity pTarget, int pTimeout = 5000);
	}

}